hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -581,6 +682,7 @@ onOpenClientClose st =
 --
 -- __Transition__: 'OpenState' → 'ClosedState'
 onOpenChainCloseTx ::
+  Monoid (UTxOType tx) =>
```

Comment:
Would it not make sense to move that constraint up in the `IsTx` class?

Reply:
It is required by the `getSnapshot` and adding this constraint further up doesn't help us. Perhaps we can refactor this later since any general `UTxO` should have a `Monoid` instance.